### PR TITLE
Add lease revoke implementation for mock.

### PR DESCRIPTION
**Pull request by ActivePeter**
Add lease revoke implementation for mock.

### CODE REVIEW

File: mock-etcd/src/mock_etcd.rs [hunk 1]
```diff
@@ -890,19 +872,20 @@ impl Lease for MockEtcd {
...
-                        let &mut (_, ref mut need_renew) = inner
-                            .lease_timeout_need_renew
+                        let &mut (_, ref mut need_renew, ref close) = inner
+                            .lease_state
                             .get_mut(&lease_id)
                             .unwrap_or_else(|| panic!("lease id not found: {lease_id}"));
                         if *need_renew {
                             timecount = timeout_sec;
                             *need_renew = false;
+                        } else if *close {
```
Comment (xiaguan) [inline mock-etcd/src/mock_etcd.rs]: Should we handle close before renew?

File: mock-etcd/src/mock_etcd.rs [hunk 2]
```diff
@@ -1563,10 +1561,46 @@ mod test {
...
+                client
+                    .lease()
+                    .revoke(EtcdLeaseRevokeRequest::new(lease_id))
+                    .await
+                    .unwrap();
+
+                let lock_start = timestamp();
+                let _res = client
+                    .lock()
+                    .lock(EtcdLockRequest::new(lock_key, lease_id))
```
Comment (xiaguan) [inline mock-etcd/src/mock_etcd.rs]: Should we use another lease_id?

File: mock-etcd/src/mock_etcd.rs (hunk 1 above)
Comment (ActivePeter) [reply mock-etcd/src/mock_etcd.rs]: I'll change it.

Comment (ActivePeter) [reply mock-etcd/src/mock_etcd.rs]: Yeah. That's right.

File: mock-etcd/src/mock_etcd.rs (hunk 2 above)
Comment (ActivePeter) [reply mock-etcd/src/mock_etcd.rs]: Yeah, that's right.

Comment (ActivePeter) [reply mock-etcd/src/mock_etcd.rs]: Yeah, that's right.

**Review by xiaguan — APPROVED** [review]

**Pull request merged**